flaky test `TestCTE.Concurrent`

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

Run the CI on PR https://github.com/pingcap/tiflash/pull/10372

### 2. What did you expect to see? (Required)

All unittests passed

### 3. What did you see instead (Required)

https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftiflash%2Fpull_unit_next_gen/detail/pull_unit_next_gen/313/pipeline/

https://do.pingcap.net/jenkins/blue/rest/organizations/jenkins/pipelines/pingcap/pipelines/tiflash/pipelines/pull_unit_next_gen/runs/313/nodes/289/steps/292/log/?start=0

```
[0;33mNote: Google Test filter = TestCTE.Concurrent
[m[0;32m[==========] [mRunning 1 test from 1 test case.
[0;32m[----------] [mGlobal test environment set-up.
[0;32m[----------] [m1 test from TestCTE
[0;32m[ RUN      ] [mTestCTE.Concurrent
/home/jenkins/agent/workspace/tiflash-build-common/tiflash/dbms/src/Operators/tests/gtest_cte.cpp:289: Failure
Expected equality of these values:
  blocks.size()
    Which is: 2620
  total_block_num
    Which is: 2621
[2025/08/26 18:30:29.597 +08:00] [ERROR] [Exception.cpp:96] ["std::exception. Code: 1001, type: testing::AssertionException, e.what() = /home/jenkins/agent/workspace/tiflash-build-common/tiflash/dbms/src/Operators/tests/gtest_cte.cpp:289: Failure\nExpected equality of these values:\n  blocks.size()\n    Which is: 2620\n  total_block_num\n    Which is: 2621"] [source="virtual void DB::tests::TestCTE_Concurrent_Test::TestBody()"] [thread_id=1]
/home/jenkins/agent/workspace/tiflash-build-common/tiflash/dbms/src/Operators/tests/gtest_cte.cpp:318: Failure
Failed
[0;31m[  FAILED  ] [mTestCTE.Concurrent (642 ms)
[0;32m[----------] [m1 test from TestCTE (642 ms total)

[0;32m[----------] [mGlobal test environment tear-down
[0;32m[==========] [m1 test from 1 test case ran. (643 ms total)
[0;32m[  PASSED  ] [m0 tests.
[0;31m[  FAILED  ] [m1 test, listed below:
[0;31m[  FAILED  ] [mTestCTE.Concurrent

 1 FAILED TEST
```

### 4. What is your TiFlash version? (Required)
master

--- a/dbms/src/Operators/tests/gtest_cte.cpp
+++ b/dbms/src/Operators/tests/gtest_cte.cpp
@@ -226,6 +226,9 @@ void concurrentTest()
             cte->sourceExit();
     };
 
+    std::vector<size_t> exit_num_for_each_partition;
+    exit_num_for_each_partition.resize(EXPECTED_SINK_NUM, 0);
+
     auto sink_func = [&](size_t sink_idx, size_t partition_idx) {
         std::uniform_int_distribution<size_t> di1(1, 10);
         std::uniform_int_distribution<size_t> di2(10, 50);
@@ -243,7 +246,8 @@ void concurrentTest()
                 next_sink_idx = next_sink_idxs[sink_idx]++;
                 if unlikely (next_sink_idx >= sink_blocks[sink_idx].size())
                 {
-                    if (partition_idx == 0)
+                    ++exit_num_for_each_partition[sink_idx];
+                    if (exit_num_for_each_partition[sink_idx] == PARTITION_NUM)
                         cte->sinkExit<true>();
                     break;
                 }
@@ -312,7 +316,7 @@ void concurrentTest()
 TEST_F(TestCTE, Concurrent)
 try
 {
-    for (size_t i = 0; i < 5; i++)
+    for (size_t i = 0; i < 10; i++)
         concurrentTest();
 }
 CATCH

